server/src/internalClusterTest/java/org/opensearch/remotestore/multipart/mocks/MockFsVerifyingBlobContainer.java

```diff
@@ -114,6 +116,23 @@ public void asyncBlobUpload(WriteContext writeContext, ActionListener<Void> comp
 
     }
 
+    @Override
+    public void readBlobAsync(String blobName, long position, long length, ActionListener<InputStream> listener) throws IOException {
+        ExecutorService executorService = Executors.newFixedThreadPool(10);
```

Comment:
No need to create a 10 thread executor just to submit one task and shut it down :) Can probably just do:

```java
new Thread(() -> {
    try {
        listener.onResponse(readBlob(blobName, position, length));
    } catch (Exception e) {
        listener.onFailure(e);
    }
})).start();
```